File: SumOfPrimes/sum.rb
# sum of primes soln in ruby for code eval by steven a dunn

def prime?(i)
  if i % 2 == 0 then return false end

  (3..i-1).step(2).each do |x|
    if i % x == 0 then return false end
  end
  true
end

# start at 2 
num = 3
sum = 2
count = 1
while count < 1000
  if prime?(num) 
    sum += num 
    count += 1
  end
  num += 2
end

puts sum

File: SumOfPrimes/sum.c
// sum of primes soln in c for codeeval by steven a dunn

#include <stdio.h>

#define TRUE 1
#define FALSE 0

int is_prime(int);

int main(int argc, char* argv[])
{
    int prime_sum = 2;
    int num = 3;
    int prime_count = 1;
    while (prime_count < 1000)
    {
        if (is_prime(num))
        {
            prime_sum += num;
            prime_count += 1;
        }
        num = num + 2;
    }
    printf("%d\n", prime_sum);
    return 0;
}

int is_prime(int n)
{
    if (n % 2 == 0) { return FALSE; }

    for(int i = 3; i < n; i = i + 2)
        if (n % i == 0) { return FALSE; }

    return TRUE;
}